src/evidently/ui/api/projects.py

```diff
@@ -28,18 +29,24 @@
 from evidently.ui.base import Project
 from evidently.ui.base import ProjectManager
 from evidently.ui.dashboards.base import DashboardPanel
+from evidently.ui.errors import NotEnoughPermissions
 from evidently.ui.type_aliases import OrgID
 from evidently.ui.type_aliases import TeamID
 from evidently.ui.type_aliases import UserID
 from evidently.utils import NumpyEncoder
 
 
-@get("/{project_id:uuid}/reports")
-async def list_reports(
+def is_authenticated(connection: ASGIConnection, _: BaseRouteHandler) -> None:
+    if not connection.scope["auth"]["authenticated"]:
+        raise NotEnoughPermissions()
+
+
+@get("/{project_id:uuid}/reports", sync_to_thread=True)
+def list_reports(
     project_id: Annotated[uuid.UUID, Parameter(title="id of project")],
-    project_manager: Annotated[ProjectManager, Dependency()],
-    log_event: Annotated[Callable, Dependency()],
-    user_id: Annotated[UserID, Dependency()],
+    project_manager: Annotated[ProjectManager, Dependency(skip_validation=True)],
```

Comment:
Same as for the collector app, validating these Pydantic models on every invokation of the endpoints might be costly.